### PR TITLE
Library Panels: Return valid panels

### PR DESCRIPTION
**Pull request by evictorero**
**What is this feature?**
When a folder containing a library panel is deleted, the get all library elements endpoint fails because it can't find the missing folder, and therefore, the library panels manage page can't be rendered.
This PR ignores invalid panels to render the library panel page with valid panels.

Related [escalation](https://github.com/grafana/support-escalations/issues/16357)

**Special notes for your reviewer:**
We have some frontend validations to not allow folder deletion when there are dependent resources:
image

When a folder is deleted through API, these validations don't get executed.
This flow will change with the migration to k8s api I don't think it worth the effort to write the validations in the legacy api. Instead, we skip invalid ilbrary panels and return a list of valid ones to be able to render the page.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/services/libraryelements/api.go [hunk 1]
```diff
@@ -382,7 +382,10 @@ func (l *LibraryElementService) filterLibraryPanelsByPermission(c *contextmodel.
 	for _, p := range elements {
 		allowed, err := l.AccessControl.Evaluate(c.Req.Context(), c.SignedInUser, ac.EvalPermission(ActionLibraryPanelsRead, ScopeLibraryPanelsProvider.GetResourceScopeUID(p.UID)))
 		if err != nil {
-			return nil, err
+			// This could fail because the folder that contains the library panel does not exist or the user doesn't have permissions to read it.
+			// We skip it instead of breaking the library panel list rendering flow and log the error.
+			l.log.Warn("Failed to evaluate permissions", "error", err)
+			continue
```
Comment (nmarrs) [inline pkg/services/libraryelements/api.go:388]: do we have any tests for this? might be good to add a edge case unit test verifying this functionality

**Review by nmarrs — APPROVED** [review]
lgtm - only thing is that we should consider adding a unit test to cover this case :)

**Review by stephaniehingtgen — APPROVED** [review]
LGTM, but agreed would be good to have a test for this

On a side note, though, there is another bug on top of this then -- we shouldn't be able to delete folders if they have something inside of them: https://github.com/grafana/grafana/blob/523758c9b347797a1e4fb9a58cc2b39b19dbf478/pkg/registry/apis/folders/register.go#L277
